api/build.gradle

```diff
@@ -1,32 +1,74 @@
 description = "reactive-streams"
-def jdkFlow = false
+def isJdk9OrGreater = false
 try {
     Class.forName("java.util.concurrent.Flow")
-    jdkFlow = true
+    isJdk9OrGreater = true
 } catch (ClassNotFoundException cnfe) {
 
 }
 
 sourceSets {
     main {
         java {
-            if (jdkFlow)
-              srcDirs = ['src/main/java', 'src/main/java9']
-            else
-              srcDirs = ['src/main/java']
+            srcDirs = ['src/main/java']
         }
     }
+    if (isJdk9OrGreater) java9 {
+        java {
+            srcDirs = ['src/main/java', 'src/main/java9']
+        }
+    }
```

Comment:
This is now a second source set, instead of an additional directory in the main source set. This allows a second compile task to target a different JVM bytecode level.